yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -119,6 +125,54 @@ def pixelize(self, dimension, data_source, field, bounds, size,
             return self._oblique_pixelize(data_source, field, bounds, size,
                                           antialias)
 
+
+    def line_plot(self, field, start_point, end_point, resolution):
+        """
+        Method for sampling grid data sets along a line in preparation for
+        one-dimensional line plots. For UnstructuredMesh, relies on a
+        sampling routine written in cython
+        """
+
+        index = self.ds.index
+        if (hasattr(index, 'meshes') and
+           not isinstance(index.meshes[0], SemiStructuredMesh)):
+            ftype, fname = field
+            if ftype == "all":
+                mesh_id = 0
+                indices = np.concatenate([mesh.connectivity_indices for mesh in index.mesh_union])
+            else:
+                mesh_id = int(ftype[-1]) - 1
+                indices = index.meshes[mesh_id].connectivity_indices
+
+            coords = index.meshes[mesh_id].connectivity_coords
+            if coords.shape[1] != end_point.size != start_point.size:
+                raise ValueError("The coordinate dimension doesn't match the "
+                                 "start and end point dimensions.")
+
+
+            offset = index.meshes[mesh_id]._index_offset
+            ad = self.ds.all_data()
+            field_data = ad[field]
+            if field_data.shape[1] == 27:
+                # hexahedral
+                mylog.warning("High order elements not yet supported, " +
+                              "dropping to 1st order.")
+                field_data = field_data[:, 0:8]
+                indices = indices[:, 0:8]
+
+            arc_length, plot_values = element_mesh_line_plot(coords, indices,
+                                                             start_point,
+                                                             end_point,
+                                                             resolution, field_data,
+                                                             index_offset=offset)
+
+            return arc_length, plot_values
+
+        else:
+            raise NotImplementedError("Currently line plotting routines have only "
+                                      "been implemented for unstructured meshes.")
```

Comment:
Would you like help with this? I bet it would be pretty easy to get working using e.g. a ray object or a covering grid. I'd honestly prefer not to merge this before yt 3.4 unless we implement this more generally